### ISSUE TITLE
Doctrine param converter option resolution differs between supports/apply

### ISSUE THREAD

**Issue by ro0NL** [sensiolabs/SensioFrameworkExtraBundle#564]
Curious ... is it intended to have options resolved strict in `apply`:

https://github.com/sensiolabs/SensioFrameworkExtraBundle/blob/29a17a724fd61f13a2894a0c6a11d35c86bfa59c/Request/ParamConverter/DoctrineParamConverter.php#L56

whereas `supports` resolves loose:

https://github.com/sensiolabs/SensioFrameworkExtraBundle/blob/29a17a724fd61f13a2894a0c6a11d35c86bfa59c/Request/ParamConverter/DoctrineParamConverter.php#L268

Basically i hit this exception pretty fast:

https://github.com/sensiolabs/SensioFrameworkExtraBundle/blob/29a17a724fd61f13a2894a0c6a11d35c86bfa59c/Request/ParamConverter/DoctrineParamConverter.php#L304-L306

when using custom options that only my converter supports and not referencing by name.

**Comment by ro0NL** [comment]
Ok. I see what i did wrong, i was using `class="App:User\User"` as well. Which triggers the doctrine converter..

setting it to any other class fixes it, as it finds my converter now.

DX issue? Feel free to close otherwise.
